From 9940665e2ef3ebdc9c56d60aded30b5dfb63adf0 Mon Sep 17 00:00:00 2001
From: "xinhui.pan" <xinhuix.pan@intel.com>
Date: Sat, 9 May 2015 12:47:09 +0800
Subject: [PATCH] Touch: fix mxt_start lost issue when restart android

when restart android, the flow related to touch are:
a) close touch input device
b) resume touch
c) input device open
This will cause mxt_start will not enable touch indeed.

Change-Id: I0ad604d71e54d94de9f396bf27b773e3588c1b4f
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35628
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1599
Signed-off-by: xinhui.pan <xinhuix.pan@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index f9256ac93993..c8eb63eeff8e 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -318,6 +318,7 @@ struct mxt_data {
 	/* Indicates whether device is updating configuration */
 	bool updating_config;
 
+	bool late_start;
 	/*debugfs interfaces for factory test*/
 #ifdef CONFIG_DEBUG_FS
 	char *debugfs_name;
@@ -3364,7 +3365,8 @@ static void mxt_reset_slots(struct mxt_data *data)
 
 static void mxt_start(struct mxt_data *data)
 {
-	if (!data->suspended || data->in_bootloader)
+	if ((!data->suspended && !data->late_start)
+			|| data->in_bootloader)
 		return;
 
 	if (data->use_regulator) {
@@ -4011,9 +4013,12 @@ static int mxt_resume(struct device *dev)
 	if (ret)
 		dev_info(&client->dev, "Can't request reset gpio\n");
 
-	if (input_dev->users)
+	if (input_dev->users) {
 		mxt_start(data);
-
+		data->late_start = false;
+	} else {
+		data->late_start = true;
+	}
 	data->suspended = false;
 	dev_info(&client->dev, "mxt_resume complete\n");
 out:
-- 
1.9.1

